packages/@react-aria/dnd/src/useDrop.ts

```diff
@@ -15,11 +15,16 @@ import * as DragManager from './DragManager';
 import {DragTypes, readFromDataTransfer} from './utils';
 import {DROP_EFFECT_TO_DROP_OPERATION, DROP_OPERATION, DROP_OPERATION_ALLOWED, DROP_OPERATION_TO_DROP_EFFECT} from './constants';
 import {DropActivateEvent, DropEnterEvent, DropEvent, DropExitEvent, DropMoveEvent, DropOperation, DragTypes as IDragTypes} from '@react-types/shared';
+import {getDnDState, setDroppedTarget} from '@react-stately/dnd';
 import {useLayoutEffect} from '@react-aria/utils';
 import {useVirtualDrop} from './useVirtualDrop';
 
 export interface DropOptions {
   ref: RefObject<HTMLElement>,
+  // TODO: maybe it would be useful for getDropOperation to also get the dragged items/dragged keys so that users can
```

Comment:
Maybe nice to have getDropOperations also get access to what items are being dragged so the user can preemptively return `cancel` for drop targets that the end user doesn't have permissions to or to prevent folders from being dragged into themselves. Bit tricky since I think we'll need to grab stuff from e.dataTransfer and that data would need to be accessed via `.getText`/`.getFiles`/`.getEntries`...